dev/com.ibm.ws.session/src/com/ibm/ws/session/http/HttpSessionImpl.java

```diff
@@ -14,64 +14,42 @@
 import java.util.Vector;
 import java.util.logging.Level;
 
-import javax.servlet.ServletContext;
-import javax.servlet.http.HttpSession;
-import javax.servlet.http.HttpSessionBindingListener;
 import javax.servlet.http.HttpSessionContext;
 
 import com.ibm.ws.session.utils.LoggingUtil;
+import com.ibm.ws.session.AbstractSessionData;
 import com.ibm.wsspi.session.ISession;
 
 /**
  * This class provides the adapted version of the ISession.
  * It simply wrappers the session and proxies any of its method calls to
  * the underlying ISession object.
- * 
- * @author dettlaff
+ * <p> 
+ * Updated for Servlet 6.0:
+ * This class now provides specific API up to servlet 5.0.  
+ * Common APIs are moved up into the AbstractHttpSession
  */
-public class HttpSessionImpl implements HttpSession {
+public class HttpSessionImpl extends AbstractSessionData {
```

Comment:
This change completely modifies the behavior of the HttpSession methods.  AbstractSessionData's logic is different than AbstractHttpSession's logic.  This needs to be changed to not change behavior.

Reply:
I see HttpSessionImpl extends AbstractSessionData extends AbstractHttpSession implements HttpSession
which should work fine with old servlet 3.1.

Reply:
@jhanders34 can I request a second look on this hierarchy change for 

public class HttpSessionImpl extends AbstractSessionData

from the surface, it looks like a behavior change.  However, the methods in AbstractSessionData do some extra checks:

1. crossoveCheck() - this is for debugging purpose and requires additional configuration to enable.
2. Check session name against "com_ibm_ejs_security_httpsession_info" and securityInfo - I cannot find where/whom may use/set this session name.  @zhang-ca thinks it is a left over from tWAS migration to OL.

then delegating to parent AbstractHttpSession's methods.

I don't see real behavior differences from the new hierarchy.

Reply:
"com_ibm_ejs_security_httpsession_info" is for Admin Console on tWAS. It wouldn't be used in Liberty.

Reply:
I am going to mark this resolved, but I am going to look at cleaning this up after this PR is merged.